lib/msf/core/payload/stager.rb

```diff
@@ -156,7 +156,14 @@ def handle_connection(conn, opts={})
     # If the stage should be sent over the client connection that is
     # established (which is the default), then go ahead and transmit it.
     if (stage_over_connection?)
-      p = generate_stage
+      uuid_raw = conn.get_once(16, 1)
```

Comment:
Absolutely agreed, this is why I wanted to point it out as something to discuss. What would a valid approach be here given that we want the UUID to have the chance to come through without blocking everything? Is this the wrong place to put it?